### DISCUSSION TITLE
supercell_size problem in ClusterSubspace

### DISCUSSION THREAD

**Discussion by mugeunson** [CederGroupHub/smol#504] [Q&A]
Hello, smol developers and others.

I'm having trouble applying the “Creating a basic Cluster Expansion” example in my case. In my case, I was wondering about the arrangement of lithium and vacancy, so only Li and vacancies are active. 

below is the code:
```import random
import numpy as np
from monty.serialization import loadfn, dumpfn
from pymatgen.core.structure import Structure
from smol.cofe import ClusterSubspace, StructureWrangler, ClusterExpansion, RegressionData

lmo_prim = loadfn('./prim.json')
lmo_entries = loadfn("entries.json")

subspace = ClusterSubspace.from_cutoffs(
    lmo_prim,
    cutoffs={2: 6, 3: 5},
    basis='sinusoid',
    supercell_size= ('Mn4+','O2-') 
)

structure = lmo_entries[1].structure

corr = subspace.corr_from_structure(structure)

print(f'The correlation vector for a structure with'
      f' composition {structure.composition} is: '
      f'\n{corr}')
```
I'm having an issue when I put supercell_size = 'O2-': "ValueError: Invalid argument for supercell_size." and put supercell_size = ('Mn4+', 'O2-'): "ZeroDivisionError: float division by zero"

I think the training data set is a problem.
My training data is a 1x2x2 supercell of primitive cell.
Is there a problem if the cells in the training data are anisotropic supercells of primitive cells?

Any help in resolving this issue would be greatly appreciated.

[entries.json](https://github.com/user-attachments/files/18021287/entries.json)
[prim.json](https://github.com/user-attachments/files/18021289/prim.json)

**Comment by qchempku2017** [comment]
Hello, @mugeunson. Typically an-isotropic supercell does not cause such problem. By a first look (haven't carefully looked so please correct me if I'm wrong), I highly suspect that is due to the behavior of StructureMathcer when computing sites. (See https://github.com/materialsproject/pymatgen/blob/0e65d35ae0b1df964639a48067b1a525979eec84/src/pymatgen/analysis/structure_matcher.py#L419C1-L446C36, and documentations of pmg: https://pymatgen.org/pymatgen.analysis.html#pymatgen.analysis.structure_matcher.StructureMatcher)

To the best of my knowledge, pymatgen's composition object may only hosts Element rather than Species. You can try changing species strings into their corresponding elements.
